### ISSUE TITLE
MannWhitneyU shouldn't do a get(double)

### ISSUE THREAD

**Issue by lbergelson** [broadinstitute/gatk#4290]

**Comment by droazen** [comment]
@jamesemery Can you fix this when you get a chance?

**Comment by droazen** [comment]
@davidbenjamin This is the issue I was talking about -- the relevant code is in `MannWhitneyU`:

```
        double sumOfAllSmallerBins = histo.get(testStatU).getValue() / 2.0;

        for (final Histogram.Bin<Double> bin : histo.values()) {
            if (bin.getId() < testStatU) sumOfAllSmallerBins += bin.getValue();
        }

        return sumOfAllSmallerBins / histo.getCount();
```

Where `testStatU` is a double. This has caused issues like the one reported in https://github.com/broadinstitute/gatk/pull/5190. I'm wondering whether the class can be refactored to not use a double as a key.

**Comment by frank-y-liu** [comment]
I quickly looked at the nature of Mann-Whitney test. Seems to me the current implementation is correct algorithmic-wise. @droazen , I guess your concern is that using double as the key to the look-up function in `map` is too fragile?

**Comment by droazen** [comment]
@frank-y-liu Yes, I think it's not only fragile, but the direct cause of the error you're seeing. @davidbenjamin has agreed to attempt a refactor of the class when he has some free time.

**Comment by droazen** [comment]
@davidbenjamin Have you had a chance to think about this one yet? Do you think it will be possible to refactor so that doubles are not used as keys? @frank-y-liu was asking about this today.

**Comment by davidbenjamin** [comment]
I just pushed a branch that might fix this.
